src/DiffTests.jl

```diff
@@ -249,6 +249,39 @@ end
 const INPLACE_ARRAY_TO_ARRAY_FUNCS = (chebyquad!, brown_almost_linear!, trigonometric!,
                                       mutation_test_1!, mutation_test_2!)
 
+############################
+# f(x::VecOrMat)::VecOrMat #
+############################
+
+diag_matrix(::Type{T}, n::Integer) where T<:Real =
+    Diagonal(LinRange(convert(T, 0.01), convert(T, 100.0), n))
+diag_matrix(x::VecOrMat) = diag_matrix(Float64, size(x, 1))
+
+hilbert_matrix(::Type{T}, n::Integer) where T<:Real =
+    [convert(T, inv(i + j - 1)) for i in 1:n, j in 1:n]
+hilbert_matrix(x::VecOrMat) = hilbert_matrix(Float64, size(x, 1))
```

Comment:
This function doesn't seem to be used anywhere in this PR.

Reply:
> This function doesn't seem to be used anywhere in this PR.

Yes, it doesn't. I was undecided which constant matrix to use, it looked like with this one the errors are higher (it is beyond my capability to track and possibly fix the source of that errors).
I can remove, if it is critical, but maybe it is better to keep it, anticipating the future extension of the test coverage.

Reply:
I generally think keeping unused code in the main branch should be avoided. Everything else looks fine though in case someone with merge rights cares about my opinion :slightly_smiling_face: .

Reply:
I agree, this should be removed before we can merge

Reply:
@oxinabox done